### PR TITLE
Resized the rotation matrix vector in the weighted outputs for the CPO postprocessor

### PR DESCRIPTION
**Pull request by srivatsvyas**
Pull Request Checklist. Please read and check each box with an X. Delete any part not applicable. Ask on the [forum](https://community.geodynamics.org/c/aspect) if you need help with any step.

Resized the rotation matrix vector while generating weighted outputs in Crystal preferred orientation postprocessor. Before the fix if the option of generating a weighted CPO was chosen in the input file, a segmentation fault was thrown. This PR looks to fix that

### For all pull requests:

* [X ] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [ X] I have tested my new feature locally to ensure it is correct.

### CODE REVIEW

File: source/postprocess/crystal_preferred_orientation.cc [hunk 1]
```diff
@@ -389,11 +389,14 @@ namespace aspect
...
 
               weighted_euler_angles.resize(n_minerals);
+              weighted_rotation_matrices.resize(n_minerals);
               std::vector<std::vector<double>> volume_fractions_grains(n_minerals,std::vector<double>(n_grains,-1.));
               for (unsigned int mineral = 0; mineral < n_minerals; ++mineral)
                 {
                   for (unsigned int i_grain = 0; i_grain < n_grains; ++i_grain)
                     {
+                      weighted_euler_angles[mineral].resize(n_grains);
+                      weighted_rotation_matrices[mineral].resize(n_grains);
```
Comment (bangerth) [inline source/postprocess/crystal_preferred_orientation.cc:399]: Out of curiosity, why was this necessary? Did you get error messages/segmentation faults without resizing?

Comment (srivatsvyas) [reply source/postprocess/crystal_preferred_orientation.cc:399]: Yes.. Had a similar issue on the add_drexpp_V3 branch and this helped so just used it here and it works.

Comment (bangerth) [reply source/postprocess/crystal_preferred_orientation.cc:399]: This points to a bigger problem. If the code previously had a bug whereby you'd get error messages without the `resize()` calls, then there are not enough tests that test this code. Would you be able to write an input file that triggers the error before, and that is fixed with the addition of the code here?

**Review by gassmoeller — APPROVED** [review]
Let's not delay this bugfix by waiting for an unrelated test. @MFraters can you later look into the testing situation for the CPO code?

Otherwise this looks good to go.

**Comment by MFraters** [comment]
Sorry, I had missed this pull request. I think we (@srivatsvyas and me) had talked about this on the last day of the hackathon when he found the issue. Great that it is merged!
